File: 11. Container With Most Water Medium/solution.java
public class solution {
    public static void main(String[] args) {
        System.out.println(maxArea(new int[]{1,8,6,2,5,4,8,3,7}));
        System.out.println(maxArea(new int[]{1,1}));
    }

    public static int maxArea(int[] height) {
        int maxAreaValue = 0;
        int left = 0;
        int right = height.length - 1;

        while (left < right) {
            int currentArea = Math.min(height[left], height[right]) * (right - left);
            maxAreaValue = Math.max(maxAreaValue, currentArea);

            if (height[left] < height[right])
                left++;
            else
                right--;
        }

        return maxAreaValue;
    }
}

File: 11. Container With Most Water Medium/solution.ts
function maxArea(height: number[]): number {
    let maxAreaValue = 0;
    let left = 0;
    let right = height.length - 1;

    while(left < right) {
        const currentArea = Math.min(height[left], height[right]) * (right - left);
        maxAreaValue = Math.max(maxAreaValue, currentArea);

        height[left] < height[right] ? ++left : --right;
    }

    return maxAreaValue;
}

console.log(maxArea([1,8,6,2,5,4,8,3,7]));
console.log(maxArea([1,1]));